src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js

```diff
@@ -1,15 +1,113 @@
+import type { Translations } from '../../flowTypes';
+import type { SelectorItems, User } from '../../../../common/types/core';
+import type { Comment as CommentType } from '../../../../common/types/feed';
+
+import messages from './messages';
+import './ActivityThread.scss';
 
 type Props = {
     children: React.Node,
+    currentUser?: User,
+    getAvatarUrl: GetAvatarUrlCallback,
+    getMentionWithQuery?: Function,
+    getUserProfileUrl?: GetProfileUrlCallback,
     hasReplies: boolean,
+    id: string,
+    mentionSelectorContacts?: SelectorItems<>,
+    onGetReplies?: (id: string) => void,
+    onReplyDelete?: Function,
+    onReplyEdit?: Function,
+    replies?: Array<CommentType>,
+    total_reply_count?: number,
+    translations?: Translations,
 };
 
-const ActivityThread = ({ children, hasReplies }: Props) => {
+const ActivityThread = ({
+    children,
+    currentUser,
+    getAvatarUrl,
+    getMentionWithQuery,
+    getUserProfileUrl,
+    hasReplies,
+    id,
+    mentionSelectorContacts,
+    onGetReplies,
+    onReplyDelete,
+    onReplyEdit,
+    replies = [],
+    total_reply_count = 0,
+    translations,
+}: Props) => {
+    const [repliesExpanded, setRepliesExpanded] = React.useState(false);
+
+    const toggleReplyMessage = repliesExpanded ? messages.hideReplies : messages.getMoreReplies;
+    const repliesToLoad = total_reply_count - 1;
+
+    const toggleReplies = () => {
+        !repliesExpanded && onGetReplies && onGetReplies(id);
+        setRepliesExpanded(previousState => !previousState);
+    };
+
+    const renderReply = (reply: CommentType) => (
+        <ActivityItem key={reply.type + reply.id} className="bcs-ActivityThread-reply" data-testid="reply">
+            <Comment
+                {...reply}
+                currentUser={currentUser}
+                getAvatarUrl={getAvatarUrl}
+                getMentionWithQuery={getMentionWithQuery}
+                getUserProfileUrl={getUserProfileUrl}
+                mentionSelectorContacts={mentionSelectorContacts}
+                onDelete={onReplyDelete}
+                onEdit={onReplyEdit}
+                permissions={{
+                    can_delete: getProp(reply.permissions, 'can_delete', false),
+                    can_edit: getProp(reply.permissions, 'can_edit', false),
+                    can_resolve: getProp(reply.permissions, 'can_resolve', false),
+                }}
+                translations={translations}
+            />
+        </ActivityItem>
+    );
+
     if (hasReplies) {
-        return <div className="bcs-ActivityThread">{children}</div>;
+        return (
+            <div className="bcs-ActivityThread">
+                {children}
+
+                {total_reply_count > 1 && (
+                    <PlainButton
+                        className={classNames('bcs-ActivityThread-button')}
+                        data-testid="bcs-ActivityThread-button"
```

Comment:
If we rendered two separate buttons for "show" and "hide" instead of overriding this single button, we could do away with the data-testid since we can search for the user visible text

Reply:
I've removed `data-testid `and kept a single button - only text inside the button is changing and I think it make more sense to keep it that way